sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -176,9 +176,10 @@ object FileSourceStrategy extends Strategy with PredicateHelper with Logging {
         l.resolve(fsRelation.dataSchema, fsRelation.sparkSession.sessionState.analyzer.resolver)
 
       // Partition keys are not available in the statistics of the files.
+      val dataColumnsWithoutPartitionCols = dataColumns.filterNot(partitionColumns.contains)
```

Comment:
>  It will possibly push down a predicate with partition column to datasource.

Oh, I see.  nit: could you leave some comments about "data columns ` dataColumns`can have partition columns... " above?

Reply:
Sure. Added.

It only happens in hive-1.2 profile, because for hive-2.3 we go for a different path to create pushed down filters. In that path, we have checked if an attribute is in the field map.

Reply:
Ah, ok. If so, the fix looks fine.

Reply:
> Sure. Added.
> 
> It only happens in hive-1.2 profile, because for hive-2.3 we go for a different path to create pushed down filters. In that path, we have checked if an attribute is in the field map.

LGTM，when  I doing that pr  https://github.com/apache/spark/pull/29406, I have thought that : from the code, seems dataColumns won't have partition col. 

Thanks for these fix.